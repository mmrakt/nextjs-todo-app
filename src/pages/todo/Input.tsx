import { TextField } from '@material-ui/core'
import AddIcon from '@material-ui/icons/Add'
import React from 'react'
import { useDispatch } from 'react-redux'

import { db } from '../../../functions/firebase'
import { handleAdd } from '../../modules/todoModule'

function Input(): any {
  const dispatch = useDispatch()
  const [text, setText] = React.useState('')

  const handleChange = (e) => setText(e.target.value)
  const handleKeyDown = (e) => {
    if (e.key === 'Enter') {
      dispatch(handleAdd(text))

      db.collection('todos').add({
        text,
        done: false,
      })
      setText('')
    }
  }
  return (
    <>
      <TextField
        data-testid="text"
        fullWidth
        type="text"
        value={text}
        placeholder="Add Todo"
        onChange={handleChange}
        onKeyDown={handleKeyDown}
      />
      <AddIcon />
    </>
  )
}

export default Input
