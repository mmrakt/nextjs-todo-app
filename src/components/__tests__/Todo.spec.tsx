import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import TodoTabPanel from '../../pages/todo/TabPanel'

import Input from '@/pages/todo/Input'
import TodoItem from '@/pages/todo/Item'
import Todo from '@/pages/todo/index'

test('Todo', () => {
  const component = renderer.create(<Todo />)
  const tree = component.toJSON()
  // snapshot作成
  expect(tree).toMatchSnapshot()
})

test('Add Todo', () => {
  const mockEvent = {
    key: 'Enter',
    target: {
      value: 'test',
    },
  }
  const wrapper = shallow(<Input />)
  wrapper.find("[data-testid='text']").simulate('keydown', mockEvent)
  // expect(props.onAdd).toBeCalled()
})

test('Toggle Todo', () => {
  const props = {
    todo: {
      key: 'test1',
      text: 'test',
      done: false,
    },
    onCheck: jest.fn(),
    onDelete: jest.fn(),
  }
  const wrapper = shallow(<TodoItem {...props} />)
  wrapper.find("[data-testid='checkbox']").simulate('change')
  expect(props.onCheck).toBeCalled()
})

test('Display Todo', () => {
  const props = {
    todos: [
      {
        key: 'test1',
        text: 'test1',
        done: false,
      },
      {
        key: 'test2',
        text: 'test2',
        done: false,
      },
      {
        key: 'test3',
        text: 'test3',
        done: false,
      },
    ],
    filter: 'TODO',
    status: 'TODO',
    onChange: jest.fn(),
    onDelete: jest.fn(),
  }
  const wrapper = shallow(<TodoTabPanel {...props} />)
  expect(wrapper.find("[data-testid='todoItem']").length).toBe(3)
})
