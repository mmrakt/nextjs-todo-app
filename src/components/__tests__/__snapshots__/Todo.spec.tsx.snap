// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo 1`] = `
<div>
  <div>
    React Todo
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    data-testid="text"
    onKeyDown={[Function]}
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Add Todo"
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
  <div
    className="MuiTabs-root"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            ALL
          </span>
        </button>
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            TODO
          </span>
          <span
            className="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
            style={Object {}}
          />
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            DONE
          </span>
        </button>
      </div>
    </div>
  </div>
  <nav
    className="MuiList-root MuiList-padding"
    hidden={true}
    role="tabpanel"
  />
  <nav
    className="MuiList-root MuiList-padding"
    hidden={false}
    role="tabpanel"
  />
  <nav
    className="MuiList-root MuiList-padding"
    hidden={true}
    role="tabpanel"
  />
</div>
`;
